rasterio/rio/warp.py

```diff
@@ -0,0 +1,197 @@
+import logging
+from math import ceil
+import click
+from cligj import format_opt
+
+from . import options
+import rasterio
+from rasterio import crs
+from rasterio.transform import Affine
+from rasterio.warp import (reproject, RESAMPLING, calculate_default_transform,
+   transform_bounds)
+
+
+logger = logging.getLogger('rio')
+
+
+@click.command(short_help='Warp a raster dataset.')
+@options.file_in_arg
+@options.file_out_arg
+@format_opt
+@options.like_file_opt
+@click.option('--dst-crs', default=None,
+              help='Target coordinate reference system.  Default: EPSG:4326')
+@options.dimensions_opt
+@options.bounds_opt
+@options.resolution_opt
+@click.option('--resampling', type=click.Choice(['nearest', 'bilinear', 'cubic',
+                'cubic_spline','lanczos', 'average', 'mode']),
+              default='nearest', help='Resampling method')
+@click.option('--threads', type=int, default=1,
+              help='Number of processing threads.')
+@click.pass_context
+def warp(
+        ctx,
+        input,
+        output,
+        driver,
+        like,
+        dst_crs,
+        dimensions,
+        bounds,
+        res,
+        resampling,
+        threads):
+    """
+    Warp a raster dataset.
+
+    The output is always overwritten.
+
+    If a template raster is provided using the --like option, the coordinate
+    reference system, affine transform, and dimensions of that raster will
+    be used for the output.  In this case --dst-crs, --bounds, --res, and
+    --dimensions options are ignored.
+
+      rio warp input.tif output.tif --like template.tif
+
+    The output coordinate reference system may be either a PROJ.4 or EPSG:nnnn
+    string,
+
```

Comment:
I think you need to use a Click `\b` directive for this block, like at https://github.com/mapbox/rasterio/blob/master/rasterio/rio/calc.py#L56